### PR TITLE
add code tabs in _overviews/scala3-book/num59

### PR DESCRIPTION
**Pull request by benluo**

### CODE REVIEW

File: _overviews/scala3-book/ca-given-using-clauses.md [hunk 1]
```diff
@@ -50,30 +69,48 @@ So the program is equivalent to the one above.
 
 In fact, since we do not need to refer to `c` in our implementation of `renderWebsite` anymore, we can even omit its name in the signature:
 
+{% tabs using2 %}
+{% tab 'Scala 3 Only' for=using2 %}
```
Comment (bishabosha) [inline _overviews/scala3-book/ca-given-using-clauses.md:73]:
```suggestion
{% tab 'Scala 3 Only' %}
```

File: _overviews/scala3-book/ca-given-using-clauses.md [hunk 2]
```diff
@@ -29,11 +35,20 @@ def renderWidget(items: List[String], c: Config): String = ???
...
+
 Let us assume that the configuration does not change throughout most of our code base.
 Passing `c` to each and every method call (like `renderWidget`) becomes very tedious and makes our program more difficult to read, since we need to ignore the `c` argument.
 
 #### Using `using` to mark parameters as contextual
+
 In Scala 3, we can mark some parameters of our methods as _contextual_.
+
+{% tabs using1 %}
+{% tab 'Scala 3 Only' for=using1 %}
```
Comment (bishabosha) [inline _overviews/scala3-book/ca-given-using-clauses.md:50]:
```suggestion
{% tab 'Scala 3 Only'  %}
```

File: _overviews/scala3-book/ca-given-using-clauses.md [hunk 3]
```diff
@@ -8,16 +8,22 @@ previous-page: ca-contextual-abstractions-intro
...
 
 ## Using Clauses
+
 When designing a system, often context information like _configuration_ or settings need to be provided to the different components of your system.
 One common way to achieve this is by passing the configuration as additional argument to your methods.
 
 In the following example, we define a case class `Config` to model some website configuration and pass it around in the different methods.
+
+{% tabs nonusing %}
+{% tab 'Scala 2 and 3' for=nonusing %}
```
Comment (bishabosha) [inline _overviews/scala3-book/ca-given-using-clauses.md:25]:
```suggestion
{% tab 'Scala 2 and 3' %}
```

File: _overviews/scala3-book/ca-given-using-clauses.md [hunk 4]
```diff
@@ -50,30 +69,48 @@ So the program is equivalent to the one above.
...
 
 For all other cases, as we will see in the next Section, there is also another way to bring contextual values into scope.
 
 ## Given Instances
+
 We have seen that we can explicitly pass arguments as contextual parameters by marking the argument section of the _call_ with `using`.
 However, if there is _a single canonical value_ for a particular type, there is another preferred way to make it available to the Scala compiler: by marking it as `given`.
 
+{% tabs given1 %}
+{% tab 'Scala 3 Only' for=given1 %}
```
Comment (bishabosha) [inline _overviews/scala3-book/ca-given-using-clauses.md:112]:
```suggestion
{% tab 'Scala 3 Only' %}
```

File: _overviews/scala3-book/ca-given-using-clauses.md [hunk 5]
```diff
@@ -84,15 +121,25 @@ given Config = config
...
+
+{% endtab %}
+{% endtabs %}
+
 In the above example we specify that whenever a contextual parameter of type `Config` is omitted in the current scope, the compiler should infer `config` as an argument.
 
 Having defined a given for `Config`, we can simply call `renderWebsite`:
 
+{% tabs given2 %}
+{% tab 'Scala 3 Only' for=given2 %}
```
Comment (bishabosha) [inline _overviews/scala3-book/ca-given-using-clauses.md:133]:
```suggestion
{% tab 'Scala 3 Only' %}
```

**Review by bishabosha — CHANGES_REQUESTED** [review]
Thank you for taking the time to do this, but please could you add the "scala 3 only" tag to the top of the page somehow, such as was done with http://localhost:4000/scala3/book/methods-main-methods.html

then going forwards we can also have a page for "implicit" (scala 2 style) and then a page that abstracts over the use cases of the two

**Comment by benluo** [comment]
> Thank you for taking the time to do this, but please could you add the "scala 3 only" tag to the top of the page somehow, such as was done with http://localhost:4000/scala3/book/methods-main-methods.html.
> 
> then going forwards we can also have a page for "implicit" (scala 2 style) and then a page that abstracts over the use cases of the two. Otherwise if we do not make separate pages, then this page should also cover the scala 2 way
> 
> @alvinj @julienrf what do you think?

I add one line as the first line.

**Review by bishabosha — APPROVED** [review]
Thanks! However please be aware that we are working on some PR to change the contextual abstractions section quite a bit

**Pull request merged**